### PR TITLE
Initial Refactor of TargetProxy to Translator Package

### PR DESCRIPTION
**Pull request by spencerhance**
Refactors TargetHttp/Https Proxy to Translator package, including SSL Policy.

### CODE REVIEW

**Comment by spencerhance** [comment]
@rramkumar1 I added another commit to handle sslPolicy.  It gets a bit tricky since we the proxy only holds a string, but we use a pointer to hold all three states.  We currently don't update the policy if the user deletes the FC or sets the policy to nil.

I'm not sure if there's a better way to handle this, or if the original commit makes more sense for now until we have a general solution for backendconfig/frontendconfig.  Ideally in the future we should delete the policy if the FC is deleted.

File: pkg/translator/translator.go [hunk 1]
```diff
@@ -37,6 +39,8 @@ import (
 type Env struct {
 	// Ing is the Ingress we are translating.
 	Ing *v1beta1.Ingress
+	// FrontendConfig is the frontendconfig associated with the Ingress
+	FrontendConfig *frontendconfigv1beta1.FrontendConfig
```
Comment (rramkumar1) [inline pkg/translator/translator.go:43]: Add TODO that this should eventually be a map of FrontendConfigs (similar to Secret)

File: pkg/translator/translator.go [hunk 2]
```diff
@@ -222,3 +226,80 @@ func (t *Translator) ToCompositeForwardingRule(env *Env, protocol namer.NamerPro
...
+
+	return proxy, sslPolicySet, nil
+}
+
+// getSslPolicyLink returns the ref to the ssl policy that is described by the
+// frontend config.  Since Ssl Policy is a *string, there are three possible I/O situations
+// 1) policy is nil -> this returns nil
+// 2) policy is an empty string -> this returns an empty string
+// 3) policy is non-empty -> this constructs the resource path and returns it
+func getSslPolicyLink(env *Env) (*string, error) {
```
Comment (rramkumar1) [inline pkg/translator/translator.go:283]: remove get

File: pkg/translator/translator_test.go [hunk 3]
```diff
@@ -383,3 +386,183 @@ func TestToForwardingRule(t *testing.T) {
...
+		},
+		{
+			desc: "frontendconfig with empty string ssl policy",
+			fc:   &frontendconfigv1beta1.FrontendConfig{Spec: frontendconfigv1beta1.FrontendConfigSpec{SslPolicy: utils.NewStringPointer("")}},
+			want: utils.NewStringPointer(""),
+		},
+	}
+
+	for _, tc := range testCases {
+		env := &Env{FrontendConfig: tc.fc}
```
Comment (rramkumar1) [inline pkg/translator/translator_test.go:558]: use t.Run and pass the description there

File: pkg/translator/translator.go (hunk 1 above)
Comment (spencerhance) [reply pkg/translator/translator.go:44]: Done

File: pkg/translator/translator.go (hunk 2 above)
Comment (spencerhance) [reply pkg/translator/translator.go]: Done

File: pkg/translator/translator_test.go (hunk 3 above)
Comment (spencerhance) [reply pkg/translator/translator_test.go]: Done

File: pkg/translator/translator_test.go [hunk 4]
```diff
@@ -383,3 +386,185 @@ func TestToForwardingRule(t *testing.T) {
...
+	for _, tc := range testCases {
+		t.Run(tc.desc, func(t *testing.T) {
+			env := &Env{FrontendConfig: tc.fc}
+			result, err := sslPolicyLink(env)
+			if err != nil {
+				t.Errorf("desc: %q, l7.getSslPolicyLink() = %v, want nil", tc.desc, err)
+			}
+
+			if !reflect.DeepEqual(result, tc.want) {
+				t.Errorf("desc: %q, l7.getSslPolicyLink() = %v, want %+v", tc.desc, result, tc.want)
```
Comment (rramkumar1) [inline pkg/translator/translator_test.go:566]: don't need this anymore

Comment (spencerhance) [reply pkg/translator/translator_test.go]: Whoops.  Fixed that and the rest of the error

**Pull request merged**